cezo_fl/shared.py

```diff
@@ -0,0 +1,39 @@
+import torch
+from typing import Sequence
+from gradient_estimators.random_gradient_estimator import RandomGradientEstimator as RGE
+
+
+def update_model_given_seed_and_grad(
+    model: torch.nn.Module,
+    optimizer: torch.optim.Optimizer,
+    grad_estimator: RGE,
+    seeds_list: Sequence[Sequence[int]],
+    grad_scalar_list: Sequence[Sequence[torch.Tensor]],
```

Comment:
prefer smaller function, i.e. less functionality for better test and combination.

So it is better to let this execute one iteration and let the caller to decide how many iterations it needs

Reply:
changed and also updated the server_test